### ISSUE TITLE
Interceptors no longer work with RxJS retry operator

### ISSUE THREAD

**Issue by geersch** [nestjs/nest#11923]
### Did you read the migration guide?

- [X] I have read the whole migration guide

### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Potential Commit/PR that introduced the regression

PR #11142

### NestJS version

9.4.2 -> 10.0.3

### Describe the regression

I maintain 2 packages `@geersch/retry` and `@geersch/nestjs-retry`. The former one offers a retry function supporting exponential backoff (powered by RxJS) and the latter allows you to use it via a NestJS interceptor.

```ts
@Controller('cats')
class CatsController {
  @Get()
  @UseInterceptors(new RetryInterceptor(FixedBackoffStrategy))
  findAll(@Headers('x-attempt') header: number): string {
    return 'all the cats';
  }
}
```

This worked fine with NestJS 9.x.x, but with NestJS 10 it breaks and the tests of `@geersch/nestjs-retry` package fail. The retry operator which is eventually piped into the `next.handle()` observable does retry X times, but it appears that the actual controller method is not invoked more than once anymore, breaking the retry mechanism.

### Minimum reproduction code

https://github.com/geersch/retry/pull/5

### Input code

I made PR where I update the dependencies to NestJS 10.0.3. 

https://github.com/geersch/retry/pull/5

The automated tests (Github actions) fail for the `@geersch/nestjs-retry` package. See `retry-interceptor.spec.ts` for the failing tests. It's just a simple controller with a few controller methods and 4 tests which all fail.

You can reproduce this by cloning this repository, switching to the `fix/nestjs10` branch and running the tests.

```
git clone git@github.com:geersch/retry.git
git checkout fix/nestjs10
yarn
```

Be sure to the `build` script (`yarn build`) from the root first before running the tests.

```
yarn workspace @geersch/nestjs-retry test --watch
```

### Expected behavior

I would expect the controller method to be retried when I pipe this into the RxJS retry operator.

**Comment by shavidze** [comment]
Yes, I think this pull request caused this issue. I'm not 100 sure but the potential reason is that the `handle` method of the handler object is still calling `nextFn(i + 1)`, but now it's not wrapped in a function.

**Comment by micalevisk** [comment]
hi @nordfjord do you mind on helping us to investigate this issue? thanks in advance.

**Comment by nordfjord** [comment]
I would suggest reverting that PR if it's causing issues with something as important as retries 😬

**Comment by nordfjord** [comment]
I believe I have a fix in #11926

**Comment by kamilmysliwiec** [comment]
Let's track this here https://github.com/nestjs/nest/pull/11926
